mixnode/src/node/http/hardware.rs

```diff
@@ -0,0 +1,61 @@
+use rocket::serde::{json::Json, Serialize};
+use sysinfo::{System, SystemExt};
+
+#[derive(Serialize)]
+#[serde(crate = "rocket::serde")]
+pub(crate) struct Hardware {
+    ram: String,
+    num_cores: usize,
+    crypto_hardware: Option<CryptoHardware>,
+}
+
+#[derive(Serialize)]
+#[serde(crate = "rocket::serde")]
+pub(crate) struct CryptoHardware {
+    aesni: bool,
+    processor: String,
+    sgx: bool,
+}
+
+/// Provides hardware information which Nym can use to optimize mixnet speed over time (memory, crypto hardware, CPU, cores, etc).
+#[get("/hardware")]
+pub(crate) fn hardware() -> Json<Option<Hardware>> {
+    Json(hardware_info())
+}
+
+/// Gives back a summary report of whatever system hardware info we can get for this platform.
+fn hardware_info() -> Option<Hardware> {
+    let crypto_hardware = hardware_info_from_cupid();
+    hardware_from_sysinfo(crypto_hardware)
+}
+
+/// Sysinfo gives back basic stuff like number of CPU cores and available memory. If available, this includes the hardware encryption
+/// extensions report
+fn hardware_from_sysinfo(crypto_hardware: Option<CryptoHardware>) -> Option<Hardware> {
+    if System::IS_SUPPORTED {
+        let mut system = System::new_all();
+        let total_memory = system.free_memory();
+        system.refresh_all();
+        let ram = format!("{}KB", total_memory);
```

Comment:
The call is for `free_memory()` here though. We either change the call or the log message. Personally, I think the total memory is more relevant here (and also less revealing of what's running on the mixnode at a certain moment)